src/serializer/ResidualFunctions.js

```diff
@@ -89,25 +102,52 @@ export class ResidualFunctions {
   prelude: Array<BabelNodeStatement>;
   factoryNameGenerator: NameGenerator;
   scopeNameGenerator: NameGenerator;
-  capturedScopeInstanceIdx: number;
-  capturedScopesArray: BabelNodeIdentifier;
-  _captureScopeAccessFunctionId: BabelNodeIdentifier;
-  serializedScopes: Map<DeclarativeEnvironmentRecord, ScopeBinding>;
+  // Null means not in an additionalFunction
+  _funcToCapturedScopeInstanceIdx: Map<FunctionValue | null, number>;
```

Comment:
Since we know the initial values of these four fields and they will be needed together anyway, I would suggest we combine them into one map:
```
_funcToScopeInfo: Map<FunctionValue | null, { scopeInstanceIndex: number, CapturedScopeArray: ...};
```

Reply:
And maybe factor out some bigger chunks into a separate file.